app/plugin_api.go

```diff
@@ -70,6 +71,19 @@ func (api *PluginAPI) UnregisterCommand(teamId, trigger string) error {
 	return nil
 }
 
+func (api *PluginAPI) ExecuteSlashCommand(commandArgs *model.CommandArgs) (*model.CommandResponse, error) {
+	user, err := api.app.GetUser(commandArgs.UserId)
+	if err != nil {
+		return nil, err
+	}
+	commandArgs.T = utils.GetUserTranslations(user.Locale)
+	response, appErr := api.app.ExecuteCommand(commandArgs)
+	if appErr != nil {
+		return response, fmt.Errorf(appErr.Error())
```

Comment:
Why don't we just return `appErr` directly here? Also, I think `err` is already a `*model.AppError`, so we can even reuse the variable name.

Reply:
Good point. I think I was just trying to handle appErr appropriately but the null checks work fine. Changed them all to `appErr`